src/components/HeadCellMenu/MenuList/__tests__/MenuGroup.test.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { fireEvent, render, screen } from '@testing-library/react';
+import MenuGroup, { interceptClickOnMenuItems } from '../MenuGroup';
+import { MenuItemGroup } from '../../types';
+
+describe('MenuGroup', () => {
+  describe('interceptClickOnMenuItems', () => {
+    let menuGroups: MenuItemGroup[];
+    let cache: Record<string, any>;
+
+    test('should return proper values per menu', () => {
+      menuGroups = [
+        [{ items: [{ id: 1, icon: () => <i />, itemTitle: 'Menu#01', enabled: true }] }],
+        [{ items: [{ id: 2, icon: () => <i />, itemTitle: 'Menu#02', enabled: true }] }],
+      ];
+      expect(interceptClickOnMenuItems(menuGroups, cache)).toMatchObject(menuGroups);
```

Comment:
This feels like a hard function to test as it essentially just returns a copy of the input object right? Not really sure this adds that much value. When are you expecting this to fail?

Reply:
in this test I'm just only checking the structure of data to match the original, since i had fall into this trap as well before when i was trying to add `groupHeading` property and it was failing to satisfy typescript because of i was missing it. 

I can merge this one with the next one about `onClick`; to only check to see if the returned structure is correct